drivers/memc/memc_stm32_xspi_psram.c

```diff
@@ -287,24 +289,34 @@ static int memc_stm32_xspi_psram_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-	hxspi.Init.ClockPrescaler = prescaler;
-	hxspi.Init.MemorySize = find_msb_set(dev_cfg->memory_size) - 2;
+	hxspi->Init.ClockPrescaler = prescaler;
+	hxspi->Init.MemorySize = find_msb_set(dev_cfg->memory_size) - 2;
 
-	if (HAL_XSPI_Init(&hxspi) != HAL_OK) {
+	if (HAL_XSPI_Init(hxspi) != HAL_OK) {
 		LOG_ERR("XSPI Init failed");
 		return -EIO;
 	}
 
-	cfg.nCSOverride = HAL_XSPI_CSSEL_OVR_NCS1;
-	cfg.IOPort = HAL_XSPIM_IOPORT_1;
+#ifndef CONFIG_STM32_APP_IN_EXT_FLASH
+	/*
+	 * Do not configure the XSPIManager if running on the ext flash
+	 * since this includes stopping each XSPI instance during configuration
+	 */
+	if (hxspi->Instance == XSPI1) {
+		cfg.IOPort = HAL_XSPIM_IOPORT_1;
+	} else if (hxspi->Instance == XSPI2) {
+		cfg.IOPort = HAL_XSPIM_IOPORT_2;
+	}
+	cfg.nCSOverride = HAL_XSPI_CSSEL_OVR_DISABLED;
 
-	if (HAL_XSPIM_Config(&hxspi, &cfg, HAL_XSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+	if (HAL_XSPIM_Config(hxspi, &cfg, HAL_XSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
 		LOG_ERR("XSPIMgr Init failed");
 		return -EIO;
 	}
+#endif
```

Comment:
Nit: prefer
```suggestion
	if (!IS_ENABLED(CONFIG_STM32_APP_IN_EXT_FLASH)) {
		/*
		 * Do not configure the XSPIManager if running on the ext flash
		 * since this includes stopping each XSPI instance during configuration
		 */
		XSPIM_CfgTypeDef cfg = {0}; //[1]

		if (hxspi->Instance == XSPI1) {
			cfg.IOPort = HAL_XSPIM_IOPORT_1;
		} else if (hxspi->Instance == XSPI2) {
			cfg.IOPort = HAL_XSPIM_IOPORT_2;
		}
		cfg.nCSOverride = HAL_XSPI_CSSEL_OVR_DISABLED;

		if (HAL_XSPIM_Config(hxspi, &cfg, HAL_XSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
			LOG_ERR("XSPIMgr Init failed");
			return -EIO;
		}
	}
```
Note proposal `[1]` to move variable declaration in this block (I'm not sure if it is necessary though, i.e. if GCC will flag as unused because the `if()` block is dead code)

Reply:
Done